importexport/musicxml/exportxml.cpp

```diff
@@ -2329,21 +2329,27 @@ static void fermatas(const QVector<Element*>& cra, XmlWriter& xml, Notations& no
             tagName += QString(" type=\"%1\"").arg(a->placement() == Placement::ABOVE ? "upright" : "inverted");
             tagName += color2xml(a);
             SymId id = a->symId();
-            if (id == SymId::fermataAbove || id == SymId::fermataBelow)
+            if (id == SymId::fermataAbove || id == SymId::fermataBelow) {
                   xml.tagE(tagName);
-            // MusicXML does not support the very short fermata nor short fermata (Henze),
-            // export as short fermata (better than not exporting at all)
-            else if (id == SymId::fermataShortAbove || id == SymId::fermataShortBelow
-                     || id == SymId::fermataShortHenzeAbove || id == SymId::fermataShortHenzeBelow
-                     || id == SymId::fermataVeryShortAbove || id == SymId::fermataVeryShortBelow)
+            }
+            else if (id == SymId::fermataShortAbove || id == SymId::fermataShortBelow) {
                   xml.tag(tagName, "angled");
-            // MusicXML does not support the very long fermata  nor long fermata (Henze),
-            // export as long fermata (better than not exporting at all)
-            else if (id == SymId::fermataLongAbove || id == SymId::fermataLongBelow
-                     || id == SymId::fermataLongHenzeAbove || id == SymId::fermataLongHenzeBelow
-                     || id == SymId::fermataVeryLongAbove || id == SymId::fermataVeryLongBelow)
+            }
+            else if (id == SymId::fermataLongAbove || id == SymId::fermataLongBelow) {
                   xml.tag(tagName, "square");
             }
+            else if (id == SymId::fermataVeryShortAbove || id == SymId::fermataVeryShortBelow) {
+                  xml.tag(tagName, "double-angled");
+            }
+            else if (id == SymId::fermataVeryLongAbove || id == SymId::fermataVeryLongBelow) {
+                  xml.tag(tagName, "double-square");
+            }
+            else if (id == SymId::fermataLongHenzeAbove || id == SymId::fermataLongHenzeBelow) {
+                  xml.tag(tagName, "double-dot");
+            }
+            else if (id == SymId::fermataShortHenzeAbove || id == SymId::fermataShortHenzeBelow) {
+                  xml.tag(tagName, "half-curve");
+            }
       }
```

Comment:
A `}` is missing here, therefore the build fails.
I guess it was the strange code style that confused you...

You could change it to the following (this way it also fits the best with the code style):
```cpp
static void fermatas(const QVector<Element*>& cra, XmlWriter& xml, Notations& notations)
      {
      for (const Element* e : cra) {
            if (!e->isFermata())
                  continue;
            const Fermata* a = toFermata(e);
            notations.tag(xml);
            QString tagName = "fermata";
            tagName += QString(" type=\"%1\"").arg(a->placement() == Placement::ABOVE ? "upright" : "inverted");
            tagName += color2xml(a);
            SymId id = a->symId();
            if (id == SymId::fermataAbove || id == SymId::fermataBelow)
                  xml.tagE(tagName);
            else if (id == SymId::fermataShortAbove || id == SymId::fermataShortBelow)
                  xml.tag(tagName, "angled");
            else if (id == SymId::fermataLongAbove || id == SymId::fermataLongBelow)
                  xml.tag(tagName, "square");
            else if (id == SymId::fermataVeryShortAbove || id == SymId::fermataVeryShortBelow)
                  xml.tag(tagName, "double-angled");
            else if (id == SymId::fermataVeryLongAbove || id == SymId::fermataVeryLongBelow)
                  xml.tag(tagName, "double-square");
            else if (id == SymId::fermataLongHenzeAbove || id == SymId::fermataLongHenzeBelow)
                  xml.tag(tagName, "double-dot");
            else if (id == SymId::fermataShortHenzeAbove || id == SymId::fermataShortHenzeBelow)
                  xml.tag(tagName, "half-curve");
            }
      }
```

Reply:
I got confused, because I introduced this change originally to the master branch, but it should be 3.x. So I just copy-pasted the lines without thinking …

Reply:
Ah, I see. Now it's OK!